src/main/scala/org/apache/spark/deploy/armada/Config.scala

```diff
@@ -54,6 +54,74 @@ private[spark] object Config {
+    * annotations, and other Kubernetes-specific configurations.
+    *
+    * Supported template sources:
+    *   - File URI: "[file:/]//absolute/path/to/template.yaml"
+    *   - HTTP/HTTPS: "https://config-server.example.com/spark-template.json"
+    *
+    * Example template content (JSON): { "queue": "default", "job_set_id": "spark-jobs",
+    * "job_request_items": [{ "priority": 1.0, "namespace": "spark-production", "labels": { "team":
+    * "data-engineering", "environment": "production" }, "pod_specs": [{ "containers": [{ "name":
+    * "spark-driver", "resources": { "requests": {"cpu": "2", "memory": "4Gi"}, "limits": {"cpu":
+    * "4", "memory": "8Gi"} } }] }] }] }
+    */
+  val ARMADA_JOB_TEMPLATE: OptionalConfigEntry[String] =
+    ConfigBuilder("spark.armada.jobTemplate")
+      .doc(
+        "URL or file path to a job template YAML/JSON file. " +
+          "Supports local files (with or without file:// prefix), " +
+          "HTTP/HTTPS URLs. The template should contain " +
+          "JobSubmitRequest configuration in JSON or YAML format."
+      )
+      .stringConf
+      .checkValue(
+        path => if (path.isEmpty) false else isValidFilePath(path),
+        "Must be a valid local file path, file://, http://, https:// URL"
+      )
+      .createOptional
+
+  val ARMADA_DRIVER_JOB_ITEM_TEMPLATE: OptionalConfigEntry[String] =
+    ConfigBuilder("spark.armada.driver.jobItemTemplate")
+      .doc(
+        "URL or file path to a job item template YAML/JSON file for the driver. " +
+          "Supports local files (with or without file:// prefix), HTTP/HTTPS URLs. " +
+          "The template should contain a JobRequestItem configuration in JSON or YAML format."
+      )
+      .stringConf
+      .checkValue(
+        path => if (path.isEmpty) false else isValidFilePath(path),
+        "Must be a valid local file path, file://, http://, https:// URL"
+      )
+      .createOptional
+
+  val ARMADA_EXECUTOR_JOB_ITEM_TEMPLATE: OptionalConfigEntry[String] =
+    ConfigBuilder("spark.armada.executor.jobItemTemplate")
+      .doc(
+        "URL or file path to a job item template YAML/JSON file for executors. " +
+          "Supports local files (with or without file:// prefix), HTTP/HTTPS URLs. " +
+          "The template should contain a JobRequestItem configuration in JSON or YAML format."
+      )
+      .stringConf
+      .checkValue(
+        path => if (path.isEmpty) false else isValidFilePath(path),
+        "Must be a valid local file path, file://, http://, https:// URL"
+      )
+      .createOptional
+
+  private def isValidFilePath(path: String): Boolean = {
+    val normalizedPath = path.toLowerCase
+    normalizedPath.startsWith("file://") ||
+    normalizedPath.startsWith("http://") ||
+    normalizedPath.startsWith("https://") ||
+    (!normalizedPath.contains("://") && path.nonEmpty) // Local file without protocol
```

Comment:
allows relative file paths, is this expected?

Reply:
I guess so, do you see any reason not to allow them?

Reply:
nope, I am fine with it, just raising attention on the inconsistency with above comment